src/rust/catnap/win/error.rs

```diff
@@ -0,0 +1,256 @@
+        WinSock::WSAENOTSOCK => libc::ENOTSOCK,
+        WinSock::WSAEDESTADDRREQ => libc::EDESTADDRREQ,
+        WinSock::WSAEMSGSIZE => libc::EMSGSIZE,
+        WinSock::WSAEPROTOTYPE => libc::EPROTOTYPE,
+        WinSock::WSAENOPROTOOPT => libc::ENOPROTOOPT,
+        WinSock::WSAEPROTONOSUPPORT => libc::EPROTONOSUPPORT,
+        WinSock::WSAEOPNOTSUPP => libc::EOPNOTSUPP,
+        WinSock::WSAEAFNOSUPPORT => libc::EAFNOSUPPORT,
+        WinSock::WSAEADDRINUSE => libc::EADDRINUSE,
+        WinSock::WSAEADDRNOTAVAIL => libc::EADDRNOTAVAIL,
+        WinSock::WSAENETDOWN => libc::ENETDOWN,
+        WinSock::WSAENETUNREACH => libc::ENETUNREACH,
+        WinSock::WSAENETRESET => libc::ENETRESET,
+        WinSock::WSAECONNABORTED => libc::ECONNABORTED,
+        WinSock::WSAECONNRESET => libc::ECONNRESET,
+        WinSock::WSAENOBUFS => libc::ENOBUFS,
+        WinSock::WSAEISCONN => libc::EISCONN,
+        WinSock::WSAENOTCONN => libc::ENOTCONN,
+        WinSock::WSAETIMEDOUT => libc::ETIMEDOUT,
+        WinSock::WSAECONNREFUSED => libc::ECONNREFUSED,
+        WinSock::WSAELOOP => libc::ELOOP,
+        WinSock::WSAENAMETOOLONG => libc::ENAMETOOLONG,
+        WinSock::WSAEHOSTUNREACH => libc::EHOSTUNREACH,
+        WinSock::WSAENOTEMPTY => libc::ENOTEMPTY,
+        WinSock::WSAESOCKTNOSUPPORT => libc::EPROTONOSUPPORT,
+        WinSock::WSAEPFNOSUPPORT => libc::EPROTONOSUPPORT,
+
+        // Winsock errors with errno equivalents missing from Rust libc
+        WinSock::WSAESHUTDOWN => libc::EINVAL,       /*libc::ESHUTDOWN*/
+        WinSock::WSAETOOMANYREFS => libc::EINVAL,    /*libc::ETOOMANYREFS*/
+        WinSock::WSAEHOSTDOWN => libc::EHOSTUNREACH, /*libc::EHOSTDOWN*/
+        WinSock::WSAEPROCLIM => libc::ENOMEM,        /*libc::EPROCLIM*/
+        WinSock::WSAEUSERS => libc::ENOMEM,          /*libc::EUSERS*/
+        WinSock::WSAEDQUOT => libc::ENOSPC,          /*libc::EDQUOT*/
+        WinSock::WSAESTALE => libc::EINVAL,          /*libc::ESTALE*/
+        WinSock::WSAEREMOTE => libc::EINVAL,         /*libc::EREMOTE*/
+
+        // Winsock errors without direct errno equivalence.
+        //
+        // System state errors
+        WinSock::WSASYSNOTREADY => libc::EFAULT,
+        WinSock::WSAVERNOTSUPPORTED => libc::EFAULT,
+        WinSock::WSANOTINITIALISED => libc::ENODEV,
+        WinSock::WSAEPROVIDERFAILEDINIT => libc::ENODEV,
+        WinSock::WSASERVICE_NOT_FOUND => libc::ENODEV,
+        WinSock::WSATYPE_NOT_FOUND => libc::ENODEV,
+
+        // Operation state errors
+        WinSock::WSATRY_AGAIN => libc::EAGAIN,
+        WinSock::WSA_E_CANCELLED => libc::ECANCELED,
+        WinSock::WSANO_RECOVERY => libc::EFAULT,
+        WinSock::WSAEDISCON => libc::ENOTCONN,
+
+        // Condition/invariant violation
+        WinSock::WSASYSCALLFAILURE => libc::EFAULT,
+
+        /* The following codes are also represented in WIN32_ERROR codes and will not be checked here:
+        WSA_INVALID_HANDLE is ERROR_INVALID_HANDLE
```

Comment:
Why are these values not checked for here? It would be good to add the reasoning to the comment.

Another option is to completely remove these commented err codes if these may not be needed in the near future.